mlir/include/mlir/Dialect/Rock/Transforms/RockMultibuffer.h

```diff
@@ -16,14 +16,24 @@
 namespace mlir {
 
 namespace rock {
-FailureOr<SmallVector<Value>> multiBuffer(RewriterBase &rewriter,
-                                          rock::GpuAllocOp allocOp,
-                                          unsigned multiplier,
-                                          bool skipOverrideAnalysis);
-
-FailureOr<SmallVector<Value>> multiBuffer(rock::GpuAllocOp allocOp,
-                                          unsigned multiplier,
-                                          bool skipOverrideAnalysis);
+FailureOr<SmallVector<rock::GpuAllocOp>> multiBuffer(RewriterBase &rewriter,
```

Comment:
Nit: `LogicalResult` and `SmallVectorImpl<GpuAllocOp>&`  to make it more obvious this is allocated on the caller's stack?